### ISSUE TITLE
warning +34 exposes #row with private types

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7438]
**Original bug ID:** 7438
**Reporter:** @avsm
**Assigned to:** @alainfrisch
**Status:** resolved (set by @alainfrisch on 2017-02-20T09:35:17Z)
**Resolution:** fixed
**Priority:** normal
**Severity:** major
**Version:** 4.04.0
**Target version:** 4.05.0 +dev/beta1/beta2/beta3/rc1
**Fixed in version:** 4.05.0 +dev/beta1/beta2/beta3/rc1
**Category:** typing
**Related to:** #6985
**Monitored by:** @gasche

## Bug description
```
# #warnings "+34";;
# module type S = sig type t = private [> `Foo] end;;
module type S = sig type t = private [> `Foo ] end
# module type X = sig type t = private [> `Foo | `Bar] include S with type t := t end;;
Warning 34: unused type t#row.
module type X = sig type t = private [> `Bar | `Foo ] end
```
The `t#row` should not expose the row variable here, and the warning+34 shouldn't be triggered.

## Additional information

Found by Thomas Gazagnaire during https://github.com/mirage/mirage/pull/743

**Comment by vicuna** [comment]
**Comment author:** @xavierleroy

@garrigue could you please have a look?  Thanks!

**Comment by vicuna** [comment]
**Comment author:** @garrigue

Since this is an unused warning problem, I suppose Alain knows best what to do.
Namely, the warning shouldn't be triggered when the name of the type ends with "#row", but I'm not sure where it is best done.
Btype has utility functions is_row_name and is_constr_row to help.

**Comment by vicuna** [comment]
**Comment author:** @xavierleroy

@frisch could you please have a look before 4.05 is released?  Thank you.

**Comment by vicuna** [comment]
**Comment author:** @alainfrisch

Looking at it.

**Comment by vicuna** [comment]
**Comment author:** @alainfrisch

Fixed by 0855920c24c7ff16b6aefeb39368e11073e8d137 on trunk, cherry-picked as f29cadc33b4ac0581c955191c9c9d5d1e4bbd009 on 4.05.
